controllers/consoleplugin.go

```diff
@@ -0,0 +1,405 @@
+	gitopsPluginSvcName          = gitopsPluginName + "-service"
+	proxyAlias                   = "gitops"
+	pluginImageEnv               = "GITOPS_CONSOLE_PLUGIN_IMAGE"
+	servicePort                  = 9001
+	pluginServingCertName        = "console-serving-cert"
+	kubeAppLabelApp              = "app"
+	kubeAppLabelComponent        = "app.kubernetes.io/component"
+	kubeAppLabelInstance         = "app.kubernetes.io/instance"
+	kubeAppLabelPartOf           = "app.kubernetes.io/part-of"
+	kubeAppLabelRuntimeNamespace = "app.kubernetes.io/runtime-namespace"
+	httpdConfigMapName           = "httpd-cfg"
+)
+
+func pluginDeployment() *appsv1.Deployment {
+	consolePluginImage := os.Getenv(pluginImageEnv)
+	if consolePluginImage == "" {
+		consolePluginImage = "quay.io/redhat-developer/gitops-console-plugin"
+	}
+	podSpec := corev1.PodSpec{
+		Containers: []corev1.Container{
+			{
+				Name:            gitopsPluginName,
+				Image:           consolePluginImage,
+				ImagePullPolicy: corev1.PullAlways,
+				Ports: []corev1.ContainerPort{
+					{
+						Name:          "http",
+						Protocol:      corev1.ProtocolTCP,
+						ContainerPort: servicePort,
+					},
+				},
+				VolumeMounts: []corev1.VolumeMount{
+					{
+						Name:      pluginServingCertName,
+						ReadOnly:  true,
+						MountPath: "/etc/httpd-ssl/certs/tls.crt",
+						SubPath:   "tls.crt",
+					},
+					{
+						Name:      pluginServingCertName,
+						ReadOnly:  true,
+						MountPath: "/etc/httpd-ssl/private/tls.key",
+						SubPath:   "tls.key",
+					},
+					{
+						Name:      httpdConfigMapName,
+						ReadOnly:  true,
+						MountPath: "/etc/httpd-cfg/httpd.conf",
+						SubPath:   "httpd.conf",
+					},
+				},
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceMemory: resourcev1.MustParse("128Mi"),
+						corev1.ResourceCPU:    resourcev1.MustParse("250m"),
+					},
+					Limits: corev1.ResourceList{
+						corev1.ResourceMemory: resourcev1.MustParse("256Mi"),
+						corev1.ResourceCPU:    resourcev1.MustParse("500m"),
+					},
+				},
+			},
+		},
+		Volumes: []corev1.Volume{
+			{
+				Name: pluginServingCertName,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName:  pluginServingCertName,
+						DefaultMode: pointer.Int32Ptr(420),
+					},
+				},
+			},
+			{
+				Name: httpdConfigMapName,
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: httpdConfigMapName,
+						},
+						DefaultMode: pointer.Int32Ptr(420),
+					},
+				},
+			},
+		},
+		RestartPolicy: corev1.RestartPolicyAlways,
+		DNSPolicy:     corev1.DNSClusterFirst,
+	}
+
+	template := corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				kubeAppLabelApp: gitopsPluginName,
+			},
+		},
+		Spec: podSpec,
+	}
+	var replicas int32 = 1
+	return &appsv1.Deployment{
```

Comment:
Can this be split into different functions ?

podSpec := getPluginPodSpec()
getPluginDeployment(podSpec)